mod/metric_answer_lookup/lib/answer.rb

```diff
@@ -8,19 +8,35 @@ class Answer < ActiveRecord::Base
   include Validations
   include EntryFetch
 
-  validates :answer_id, numericality: { only_integer: true }, presence: true
+  validates :answer_id, numericality: { only_integer: true }, presence: true,
+            unless: :virtual?
   validate :must_be_an_answer, :card_must_exist, :metric_must_exit
 
   def card_column
     :answer_id
   end
 
+  def card
+    return super if answer_id
+    @card ||= virtual_answer_card
+  end
+
+  def virtual_answer_card name=nil, val=nil
+    name ||= [record_name, year.to_s]
+    val ||= value
+    Card.new(name: name, type_id: Card::MetricValueID).tap do |card|
+      card.define_singleton_method(:value) { val }
+      card.define_singleton_method(:value_card) do
+        Card.new name: [name, :value], content: val
+      end
```

Comment:
we might consider adding all these virtual card methods to a module and having the singleton_class include it. My hunch is that it would be faster (performance-wise)